opnode/rollup/driver/step.go

```diff
@@ -142,11 +141,11 @@ func (d *outputImpl) step(ctx context.Context, l2Head eth.BlockID, l2Finalized e
 	if err != nil {
 		return l2Head, fmt.Errorf("failed to create l1InfoTx: %w", err)
 	}
-	receipts, err := d.dl.FetchReceipts(fetchCtx, l1Input[0])
+	receipts, err := d.dl.FetchReceipts(fetchCtx, l1Input[0], l1Info.ReceiptHash())
```

Comment:
I don't think this requires an interface change unless you wanted to be explicit because `FetchReceipts` internally has the block in context but its not currently assigned to a variable, see
https://github.com/ethereum-optimism/optimistic-specs/blob/b6f8fa6ad0a422e8772e2bcf2979d142fdd25a4b/opnode/l1/source.go#L81

Reply:
We had separated the Downloader calls, as we may not pull all data from one place in the future (e.g. blobs). I'll look if this can be optimized when working on #187